### DISCUSSION TITLE
Earning Twitch Channel Points

### DISCUSSION THREAD

**Discussion by Jademalo** [streamlink/streamlink#5375] [Q&A] [answered]
### Description

I'm not sure if this is the best place to ask since I know due to various reasons a direct fix is out of the direct scope of Streamlink, but it's also a direct consequence of Streamlink so I don't know a better place to ask.

https://github.com/streamlink/streamlink/issues/3297 was created a few years ago asking if it would be possible to earn channel points while watching through Streamlink, but was denied due to twitch login being removed.

Initially I thought `--twitch-api-header=Authorization=OAuth` could solve this problem, but it still doesn't allow a user to earn channel points while watching a stream through Streamlink. I also attempted adding `--twitch-access-token-param=playerType=site`, but that also didn't solve the issue. 
Opening a logged in popout chat window doesn't seem to allow for earning channel points either, and the only way it seems to work is if the actual stream is open in a browser.

Does anyone know what is actually required for earning channel points on a given stream?
I'm wondering if there's a header that could be sent as an argument that would trick the site into thinking I'm present and therefore allowing the earning of points, or if it would be possible to write a browser plugin or something to earn them simply having the popout chat open.

If anyone has any ideas I would massively appreciate it, it's hardly important but it would be nice to be able to participate in point related activities while still using Streamlink.

**Comment by bastimeyer** [comment, marked as answer]
This is completely out of scope of the Streamlink project and its goals. You'll have to ask somewhere else.

If something like that was ever going to get implemented, then this would make Streamlink an even bigger target for their bot detection because people would abuse it for earning those highly idiotic points (to put it lightly).

**Reply by Jademalo** [reply]
Alright, no worries, that's understandable. 
I don't disagree that the points are idiotic, but it does suck to not be able to join in lol.

Is there anywhere that has more documentation on the `twitch-access-token-param` option? I can't think of anywhere else to ask that would have people as knowledgable on this subject.

**Reply by bastimeyer** [reply]
- https://github.com/streamlink/streamlink/blob/28ab1e3b2e6cef7d842a8a30ba5685260659118a/src/streamlink/plugins/twitch.py#L253-L273
- https://github.com/streamlink/streamlink/blob/28ab1e3b2e6cef7d842a8a30ba5685260659118a/src/streamlink/plugins/twitch.py#L409-L418
- https://github.com/streamlink/streamlink/blob/28ab1e3b2e6cef7d842a8a30ba5685260659118a/src/streamlink/plugins/twitch.py#L275-L286

**Reply by Jademalo** [reply]
Thanks!
